src/main/java/net/openhft/chronicle/bytes/AbstractBytes.java

```diff
@@ -406,7 +406,10 @@ public byte readByte()
             return bytesStore.readByte(offset);
 
         } catch (BufferUnderflowException e) {
-            return 0;
+            if (lenient) {
+                return 0;
+            }
```

Comment:
This is a massive performance killer...

Reply:
You mean throwing and catching the exception? Agreed. But we should not be using `lenient == true` in the hot path

Reply:
...or run into EOS in the hot path